app/routes/cdo/delete/dog.js

```diff
@@ -23,18 +24,18 @@ const dogAndOwnerConfirmation = {
       return h.redirect(`${routes.viewDogDetails.get}/${pk}${extractBackNavParam(request)}`).takeover()
     }
 
-    const backLink = `${routes.deleteDog.get}/${pk}?src=${extractBackNavParam(request)}`
+    const backNav = getBackLinkToSamePage(request)
```

Comment:
But might be worth changing the var to 'backLink' since the 'backNav' is usually an object that contains backLink